sys-libs/zlib/zlib-1.2.11-r2.ebuild

```diff
@@ -0,0 +1,128 @@
+HOMEPAGE="https://zlib.net/"
+SRC_URI="https://zlib.net/${P}.tar.gz
+	http://www.gzip.org/zlib/${P}.tar.gz
+	http://www.zlib.net/current/beta/${P}.tar.gz"
+
+LICENSE="ZLIB"
+SLOT="0/1" # subslot = SONAME
+KEYWORDS="~alpha amd64 ~arm ~arm64 ~hppa ia64 ~m68k ~mips ppc ppc64 ~s390 ~sh ~sparc x86 ~amd64-fbsd ~sparc-fbsd ~x86-fbsd"
+IUSE="minizip static-libs"
+
+DEPEND="minizip? ( ${AUTOTOOLS_DEPEND} )"
+RDEPEND="abi_x86_32? (
+		!<=app-emulation/emul-linux-x86-baselibs-20130224
+		!app-emulation/emul-linux-x86-baselibs[-abi_x86_32(-)]
+	)
+	!<dev-libs/libxml2-2.7.7" #309623
+
+src_prepare() {
+	epatch "${FILESDIR}"/${PN}-1.2.11-fix-deflateParams-usage.patch
+
+	if use minizip ; then
+		cd contrib/minizip || die
+		eautoreconf
+	fi
+
+	case ${CHOST} in
+	*-mingw*|mingw*)
+		# uses preconfigured Makefile rather than configure script
+		multilib_copy_sources
+		;;
+	esac
+}
+
+echoit() { echo "$@"; "$@"; }
+
+multilib_src_configure() {
+	case ${CHOST} in
+	*-mingw*|mingw*)
+		;;
+	*)      # not an autoconf script, so can't use econf
+		local uname=$("${EPREFIX}"/usr/share/gnuconfig/config.sub "${CHOST}" | cut -d- -f3) #347167
+		echoit "${S}"/configure \
+			--shared \
+			--prefix="${EPREFIX}/usr" \
+			--libdir="${EPREFIX}/usr/$(get_libdir)" \
+			${uname:+--uname=${uname}} \
+			|| die
+		;;
+	esac
+
+	if use minizip ; then
+		local minizipdir="contrib/minizip"
+		mkdir -p "${BUILD_DIR}/${minizipdir}" || die
+		cd ${minizipdir} || die
+		ECONF_SOURCE="${S}/${minizipdir}" \
+		econf $(use_enable static-libs static)
+	fi
+}
+
+multilib_src_compile() {
+	case ${CHOST} in
+	*-mingw*|mingw*)
+		emake -f win32/Makefile.gcc STRIP=true PREFIX=${CHOST}-
+		sed \
+			-e 's|@prefix@|/usr|g' \
+			-e 's|@exec_prefix@|${prefix}|g' \
+			-e 's|@libdir@|${exec_prefix}/'$(get_libdir)'|g' \
+			-e 's|@sharedlibdir@|${exec_prefix}/'$(get_libdir)'|g' \
+			-e 's|@includedir@|${prefix}/include|g' \
+			-e 's|@VERSION@|'${PV}'|g' \
+			zlib.pc.in > zlib.pc || die
+		;;
+	*)
+		emake
+		;;
+	esac
+	use minizip && emake -C contrib/minizip
+}
+
+sed_macros() {
+	# clean up namespace a little #383179
+	# we do it here so we only have to tweak 2 files
+	sed -i -r 's:\<(O[FN])\>:_Z_\1:g' "$@" || die
+}
+
+multilib_src_install() {
+	case ${CHOST} in
+	*-mingw*|mingw*)
+		emake -f win32/Makefile.gcc install \
+			BINARY_PATH="${ED}/usr/bin" \
+			LIBRARY_PATH="${ED}/usr/$(get_libdir)" \
+			INCLUDE_PATH="${ED}/usr/include" \
+			SHARED_MODE=1
+		# overwrites zlib.pc created from win32/Makefile.gcc #620136
+		insinto /usr/$(get_libdir)/pkgconfig
+		doins zlib.pc
+		;;
+
+	*)
+		emake install DESTDIR="${D}" LDCONFIG=:
+		gen_usr_ldscript -a z
+		;;
+	esac
+	sed_macros "${ED}"/usr/include/*.h
+
+	if use minizip ; then
+		emake -C contrib/minizip install DESTDIR="${D}"
+		sed_macros "${ED}"/usr/include/minizip/*.h
+	fi
+
+	find "${ED}" -name '*.la' -delete || die
+	# 419645
+	use static-libs || find "${ED}" -name '*.a' ! -name '*.dll.a' -delete || die
```

Comment:
this is wrong, do instead
```
if use static-libs; then
	find "${D}" \( -name '*.a' -o -name '*.dll.a' \) -delete || die
fi
```

Reply:
@SoapGentoo uh... unless I'm mistaken would delete both `'*.a'` and `'*.dll.a'` files if `USE=static-libs`, which is wrong for the former (if we set `USE=static-libs` we want to keep that) and will always be wrong the latter (`'*.dll.a'` files are required (generally speaking) during the linking stage of program compilation on mingw-w64, being just a smattering of gluecode and not what one would consider a true static library)